examples/tutorials/plot_5_warm_starting.py

```diff
@@ -0,0 +1,375 @@
+
+    dry_run: bool = False
+    """quickly check a single pass"""
+
+    seed: int = 1
+    """random seed (default: 1)"""
+
+    log_interval: int = 10
+    """how many batches to wait before logging training status"""
+
+    save_model: bool = False
+    """For Saving the current Model"""
+
+    data_dir: Path = Path(os.environ.get("DATA_DIR", "data"))
+    """ Directory where the dataset should be found or downloaded."""
+
+
+class Net(nn.Sequential):
+    """Simple convnet."""
+
+    def __init__(self, n_classes: int = 10):
+        super().__init__(
+            # NOTE: `in_channels` is determined in the first forward pass
+            nn.LazyConv2d(32, 3, 1),
+            nn.ReLU(),
+            nn.Conv2d(32, 64, 3, 1),
+            nn.ReLU(),
+            nn.MaxPool2d(2),
+            nn.Dropout2d(0.25),
+            nn.Flatten(),
+            # NOTE: `in_features` is determined in the first forward pass
+            nn.LazyLinear(128),
+            nn.ReLU(),
+            nn.Dropout(0.5),
+            nn.Linear(128, n_classes),
+        )
+
+
+def train_epoch(
+    args: Args,
+    model: Net,
+    device: torch.device,
+    train_loader: DataLoader,
+    optimizer: Optimizer,
+    epoch: int,
+):
+    """Trains a model given the configuration."""
+    model.train()
+    loss_function = F.cross_entropy
+
+    pbar = tqdm.tqdm(train_loader, desc=f"Epoch {epoch}")
+    for batch_idx, (data, target) in enumerate(pbar):
+        data, target = data.to(device), target.to(device)
+        optimizer.zero_grad()
+        output = model(data)
+        loss = loss_function(output, target)
+        loss.backward()
+        optimizer.step()
+        if batch_idx % args.log_interval == 0:
+            pbar.set_postfix({"loss": loss.item()})
+            if args.dry_run:
+                break
+
+
+def test_epoch(model: Net, device: torch.device, test_loader: DataLoader) -> float:
+    """Tests the model, returning the average test loss."""
+    model.eval()
+    test_loss = 0.0
+    correct = 0
+
+    num_batches = len(test_loader.dataset)  # type: ignore
+
+    with torch.no_grad():
+        for data, target in test_loader:
+            data, target = data.to(device), target.to(device)
+            output = model(data)
+            # sum up batch loss
+            test_loss += F.nll_loss(output, target, reduction="sum").item()
+            # get the index of the max log-probability
+            pred = output.argmax(dim=1, keepdim=True)
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= num_batches
+
+    print(
+        f"\nTest set: Average loss: {test_loss:.4f}, Accuracy: {correct}/{num_batches} "
+        f"({100.0 * correct / num_batches:.0f}%)\n"
+    )
+    return test_loss
+
+
+def main(**kwargs):
+    """Main loop. Trains and then tests a model after each epoch."""
+    # Training settings
+
+    # note: could also use simple-parsing to parse the Config from the command-line:
```

Comment:
Sure!

Reply:
The note is meant for the users of the example. Enabling this would add `simple-parsing` as a dependency. I'd rather leave it as-is, if that's alrght with you.